### PR TITLE
Kp/fix datasets ids

### PR DESCRIPTION
**Pull request by kprokofi**

### CODE REVIEW

File: torchreid/engine/engine.py [hunk 1]
```diff
@@ -583,7 +583,8 @@ def test(
         top1, cur_top1, ema_top1 = [-1]*3
         should_save_ema_model = False
 
-        print('##### Evaluating test dataset) #####')
+        print('##### Evaluating test dataset #####')
+        # TO DO reduce amount of code for evaluation functions (DRY rule)
```
Comment (sovrasov) [inline torchreid/engine/engine.py]: Is it still relevant?

Comment (kprokofi) [reply torchreid/engine/engine.py]: I rewrote it a bit

File: torchreid/engine/engine.py [hunk 2]
```diff
@@ -580,15 +580,15 @@ def test(
         """
 
         self.set_model_mode('eval')
-        top1, cur_top1, ema_top1 = [-1]*3
-        should_save_ema_model = False
+        models_to_eval = list(self.models.items())
+        top1=[]
+        if (self.use_ema_decay and not lr_finder and not test_only):
+            models_to_eval.append(('EMA model', self.ema_model.module))
```
Comment (sovrasov) [inline torchreid/engine/engine.py:586]: Does `.module` work on CPU?

Comment (kprokofi) [reply torchreid/engine/engine.py:586]: Yes, it is just an attribute of the ModelEmaV2 class represented deep copy of the model. I've checked it on CPU anyway.

File: scripts/default_config.py [hunk 3]
```diff
@@ -97,6 +97,8 @@ def get_default_config():
     cfg.data.tb_log_dir = ''  # path to save tensorboard log. If empty, log will be saved to data.save_dir
     cfg.data.min_samples_per_id = 1
     cfg.data.num_sampled_packages = 1
+    cfg.data.num_classes = None # provide full number of classes if evaluation / training
+                                #  will be on part of the classes from train dataset
```
Comment (sovrasov) [inline scripts/default_config.py]: What does mean `on part`? Generally this value should be inherited from annotation (in train mode)/model (in eval script). One more option is to manually extend the list of classes in val if some of them are missing there. Also dummy extension won't solve possible indexing discrepancies between train and val. I think this feature requires some extra discussions.

Comment (kprokofi) [reply scripts/default_config.py]: Agree with you. I will rewrite it. I will inherit this information from the annotation file. 
The annotation file should always contain a full number of labels

Comment (kprokofi) [reply scripts/default_config.py]: Please, review. I've checked that it works witch VOC and OpenImages

File: torchreid/data/datasets/image/classification.py [hunk 4]
```diff
@@ -28,25 +27,14 @@ def __init__(self, root='', mode='train', **kwargs):
...
+            self.data_dir,
+        )
 
-        else:
-            classes = []
-            train, test = [], []
-
-        super().__init__(train, test, mode=mode, classes=classes, **kwargs)
+        super().__init__(data, **kwargs)
+        self.classes = classes
```
Comment (sovrasov) [inline torchreid/data/datasets/image/classification.py]: Why don't you pass classes and num_ids as args to constructor?

File: torchreid/data/datasets/image/classification.py [hunk 5]
```diff
@@ -77,33 +65,22 @@ def load_annotation(annot_path, data_dir):
...
+        data, classes = self.load_annotation(
                 self.data_provider
-            )
-            train = []
-        else:
-            classes = []
-            train, test = [], []
+                )
 
-        super().__init__(train, test, mode=mode, classes=classes,
```
Comment (sovrasov) [inline torchreid/data/datasets/image/classification.py:98]: Why `mixed_cls_heads_info` was removed from the list of the constructor args?

File: torchreid/data/datasets/image/classification.py (hunk 4 above)
Comment (kprokofi) [reply torchreid/data/datasets/image/classification.py]: good point!

Comment (kprokofi) [reply torchreid/data/datasets/image/classification.py]: done

**Review by sovrasov — APPROVED** [review]

**Pull request merged**